File: 백준/Silver/13241. 최소공배수/최소공배수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringBuilder sb = new StringBuilder();
        StringTokenizer st = new StringTokenizer(br.readLine());

        long a =  Long.parseLong(st.nextToken());
        long b = Long.parseLong(st.nextToken());

        long gcd = findGcd(a, b);
        long lcm = (a * b) / gcd;

        sb.append(lcm);

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }

    private static long findGcd(long a, long b) {
        if (b == 0) {
            return a;
        } else {
            return findGcd(b, a % b);
        }
    }
}

File: 백준/Silver/13241. 최소공배수/최소공배수.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val sb = StringBuilder()
    val st = StringTokenizer(br.readLine())

    val a = st.nextToken().toLong()
    val b = st.nextToken().toLong()

    val gcd = findGcd(a, b)
    val lcm = (a * b) / gcd

    sb.append(lcm)

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}

fun findGcd(a: Long, b: Long): Long {
    if (b.toInt() == 0) {
        return a
    } else {
        return findGcd(b, a % b)
    }
}
